### PR TITLE
docs: Add common code-review checks to PR template

### PR DESCRIPTION
**Pull request by theopolis**
Adding some common high-level "checks" we perform when coding-reviewing.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -8,3 +8,21 @@ To submit a PR please make sure to follow the next steps:
...
+- Common utilities within `./osquery/utils` are used where appropriate (avoid reinventions).
+- Modern C++11 structures and patterns are used where appropriate.
+- No memory or file descriptor leaks, please check all early-return and destructors.
+- No explicit casting, such as `return (int)my_var`, instead use `static_cast`.
+- The minimal ammount of includes are used, only include what you use.
+- Comments for methods, structures, and classes follow our common patterns.
+- `Status` and `LOG(N)` messages do not use punctuation or contractions.
+- Support for both CMake and BUCK (we are happy to help).
+- The code mostly looks and feels similar to the existing codebase.
+
```
Comment (muffins) [inline .github/PULL_REQUEST_TEMPLATE.md]: Perhaps something to the affect of `unit tests, or a minimal test plan, are required`? Or something encourage folks to submit simple unit testing/generic "I did X" test plans?

Comment (theopolis) [reply .github/PULL_REQUEST_TEMPLATE.md]: Very much agree, I think that's captured in the above "todo list"

```
- [ ] Ensure your PR contains tests for the changes you're submitting.
- [ ] Describe your changes with as much detail as you can.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -8,3 +8,21 @@ To submit a PR please make sure to follow the next steps:
...
 - [ ] Remove the text above.
+
+<!--
+
+The PR will be reviewed by an osquery committer.
+Here are some common things we look for:
+
+- The code is formatted correctly, considering using `make format_check`.
+- Common utilities within `./osquery/utils` are used where appropriate (avoid reinventions).
+- Modern C++11 structures and patterns are used where appropriate.
```
Comment (Smjert) [inline .github/PULL_REQUEST_TEMPLATE.md]: Shouldn't we strive to use Modern C++ in general?
What throws me off is that written like that Modern C++ seems bad?

C++11 is 8 years old, we compile with C++14.
I'm not saying we should use every feature of C++11/14 blindly, but surely write Modern C++, not pre-C++11 code, not C (unless for optimization or around system APIs where's necessary, but that should be very limited).

Comment (theopolis) [reply .github/PULL_REQUEST_TEMPLATE.md]: We should, but "where appropriate" handles interaction with OS/third-party libraries. There are reasonable instances where we cannot use modern structures. But I will defer to recommendations here - if you have a better suggestion on how to write this - we are saying the same thing: use modern C++ as much as possible!

Comment (Smjert) [reply .github/PULL_REQUEST_TEMPLATE.md]: Then I suggest changing with "where possible" :).
I think that's a bit stronger and making the case for "impossible" it's a bit more difficult.

**Pull request merged**